environs/interface.go

```diff
@@ -493,3 +493,12 @@ type UpgradeStep interface {
 	// Run executes the upgrade business logic.
 	Run(ctx context.ProviderCallContext) error
 }
+
+// DefaultConstraintsChecker defines an interface for checking if the default
+// constraints should be applied for the Environ provider when bootstrapping
+// the provider.
+type DefaultConstraintsChecker interface {
+	// ShouldUseDefaultConstraints returns if bootstrapping logic should use
+	// default constraints
+	ShouldUseDefaultConstraints() bool
```

Comment:
Though unlikely, I would accommodate extension of this interface by using a specific name, like:
```
ShouldApplyControllerConstraints() bool
```
This is congruent with `withDefaultControllerConstraints`.